### DISCUSSION TITLE
Who can manually trigger a workflow using workflow_dispatch

### DISCUSSION THREAD

**Discussion by itsayellow** [community/community#26053] [Actions]
<p>I’m using a workflow that is designed to be manually triggered only on <code>workflow_dispatch</code>.<br>
It also uses an action (mhausenblas/mkdocs-deploy-gh-pages) that I use a GITHUB_TOKEN with in order to publish to my gh-pages branch.</p>
<p>My question is who is authorized to initiate a manually-triggered workflow like my case above?  I assume it’s only people who have write access to the repo, but I couldn’t find any definitive answer by googling.</p>
<p>Can somebody point me to a reference for this, or explain a definitive answer?</p>
<p>thanks!</p>

**Comment by nethgato** [comment]
<p>Hey <a class="mention" href="/itsayellow">@itsayellow</a>! Welcome to the community and thanks for your post.</p>
<blockquote>
<p>My question is who is authorized to initiate a manually-triggered workflow like my case above? I assume it’s only people who have write access to the repo, but I couldn’t find any definitive answer by googling.</p>
</blockquote>
<p>Almost! There has to be an authorized token. Whether that’s an authorized PAT, or the <code>GITHUB_TOKEN</code> for the repository, an individual will have to have a token with appropriate privileges.</p>
<p>We discuss this a bit, here:</p>
<p><a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#triggering-new-workflows-using-a-personal-access-token" class="onebox" target="_blank">https://docs.github.com/en/actions/reference/events-that-trigger-workflows#triggering-new-workflows-using-a-personal-access-token</a></p>
<p>…and in more detail, here:</p>
<p><a href="https://docs.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token" class="onebox" target="_blank">https://docs.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token</a></p>
<p>Between these pages, you should have your answer, but please let us know if anything remains!</p>

**Comment by itsayellow** [comment]
<p>Thanks for the reply!  I did indeed see those pages before.</p>
<p>A personal access token makes sense to me, it’s basically a way for something else to act with my permissions.</p>
<p>Does a GITHUB_TOKEN work the same way, with a maximum of my permissions?</p>
<p>I think my confusion is in the fact that a GITHUB_TOKEN is generated automatically.  I was a little unclear in exactly what manner that happens, and what permissions (i.e. on which user’s behalf) it operates.  Is it equivalent to a personal access token of mine, if I am the one initiating the action?</p>
<p>My ultimate question is, could somebody who doesn’t have access to a repo initiate a workflow action that needs a GITHUB_TOKEN and have it run and make changes to the repo?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/itsayellow">@itsayellow</a>,</p>
<blockquote>
<p>Does a GITHUB_TOKEN work the same way, with a maximum of my permissions?</p>
</blockquote>
<p>No, the <strong>GITHUB_TOKEN</strong> does not have the full access.</p>
<p>As you can see the introduction from the docs shared by <a class="mention" href="/nethgato">@nethgato</a> about the <strong>GITHUB_TOKEN</strong>, <a href="https://docs.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token#about-the-github_token-secret" rel="nofollow noopener">the permissions of <strong>GITHUB_TOKEN</strong> are limited to the repository that contains your workflow.</a><br>
And even if in the workflow repository, the <strong>GITHUB_TOKEN</strong> also does not have full access, for example you can’t use the <strong>GITHUB_TOKEN</strong> to add, update or delete workflow files.<br>
Many APIs related to workflows require the authentication token has the ‘<strong>workflow</strong>’ scope, but the <strong>GITHUB_TOKEN</strong> does not have this scope.</p>
<p>The <strong>GITHUB_TOKEN</strong> is generated to authenticate on behalf of GitHub Actions, no matter who triggers the workflows, the permissions of the <strong>GITHUB_TOKEN</strong> is fixed. Normally, its permissions won’t be different according to different users who trigger the workflow.</p>
<blockquote>
<p>My ultimate question is, could somebody who doesn’t have access to a repo initiate a workflow action that needs a <strong>GITHUB_TOKEN</strong> and have it run and make changes to the repo?</p>
</blockquote>
<p>To trigger a workflow in a repository, the user should be a collaborator with <strong>Write</strong> permission in the repository. Normally the external users can’t trigger workflows in the repository.<br>
If the external users use a pull request to trigger workflow from the forked repository, the <strong>GITHUB_TOKEN</strong> only has <strong>read</strong> permissions for some scope.</p>

**Comment by itsayellow** [comment]
<p>Great, ok I think that answers my main concern.  Thanks!</p>
<p>In general it would be nice to clarify this in the documentation, since GITHUB_TOKEN isn’t as easy to understand as a normal secret.  At least to me. :slight_smile:</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/itsayellow">@itsayellow</a>,</p>
<blockquote>
<p>In general it would be nice to clarify this in the documentation</p>
</blockquote>
<p>I agree.<br>
You can try to share a feedback <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a> to ask improving the documentation. That will allow you to directly interact with the appropriate engineering team, and make it more convenient for the engineering team to collect and categorize your suggestions.</p>
<p>In addition, if you feel my above explanation is very helpful to you, maybe you can mark it as the solution of this topic, so that other users who have the similar questions can notice this when they are looking for an answer.</p>
